src/airdrop/airdrop.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable } from '@nestjs/common';
+import {
+  RpcProvider,
+  Account,
+  Contract,
+  GetTransactionReceiptResponse,
+  TransactionFinalityStatus,
+} from 'starknet';
+import { AirdropDto } from 'src/payment/dto/airdrop.dto';
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class AirdropService {
+  private provider: RpcProvider;
+  private account: Account;
+
+  constructor(private configService: ConfigService) {
+    this.initializeProviderAndAccount();
+  }
+
+  private initializeProviderAndAccount(): void {
+    const nodeUrl = this.configService.get<string>('NODE_URL');
+    const rpcApiKey = this.configService.get<string>('RPC_API_KEY');
+    const walletAddress = this.configService.get<string>(
+      'ARGENT_WALLET_ADDRESS',
+    );
+    const walletPrivateKey = this.configService.get<string>('ARGENT_WALLET_PK');
+
+    const apiUrl = `${nodeUrl}?apikey=${rpcApiKey}`;
+    this.provider = new RpcProvider({ nodeUrl: apiUrl });
```

Comment:
we should have a factory to create this RpcProvider so that you can have it in dependency injection